eval-config.nix

```diff
@@ -1,10 +1,75 @@
+let
+  nixDarwinVersion = builtins.fromJSON (builtins.readFile ./version.json);
+
+  checkRelease = lib:
+    # Avoid breaking configurations when the unstable Nixpkgs version
+    # rolls over.
+    #
+    # TODO: Something more refined than this would be ideal, as this
+    # still means you could be using unstable nix-darwin 25.05 with
+    # Nixpkgs 26.05, which would be unfortunate.
+    if nixDarwinVersion.isReleaseBranch then
+      lib.trivial.release == nixDarwinVersion.release
+    else
+      lib.versionAtLeast lib.trivial.release nixDarwinVersion.release;
+in
+
 { lib
 , modules
 , baseModules ? import ./modules/module-list.nix
 , specialArgs ? { }
 , check ? true
+, enableNixpkgsReleaseCheck ? true
```

Comment:
Do we want to introduce a new argument here? It's probably not a big deal, but I wonder if any users would end up disabling this for whatever reasons, if we want to keep it maybe we could rename it to something like `allowUnsupportedNixpkgsVersions`